onnx/common/tensor.h

```diff
@@ -4,6 +4,42 @@
 #pragma once
 
 #include "onnx/onnx_pb.h"
+#include <math.h>
+#include <numeric>
+#include <functional>
+
+#define APPLY_FUNCTION(f)  {                                  \
```

Comment:
looks to me `apply_binary_function_*` functions are doing similar type switch again to determine the underlying field and forward to `bin_func`, can we collapse those into here and remove another level of indirection?

Reply:
compressed